### PR TITLE
Issue/266

### PR DESCRIPTION
**Pull request by minborg**
This PR adds more than 600 tests and fixes hundreds of issues related to closed resources.

### CODE REVIEW

File: src/main/java/net/openhft/chronicle/bytes/Bytes.java [hunk 1]
```diff
@@ -181,7 +182,7 @@
      * Here is an example of field groups:
      * <pre>{@code
      *     static class Padding extends Parent {
-     *         @FieldGroup("p")
+     *         {@literal @}FieldGroup("p")
```
Comment (RobAustin) [inline src/main/java/net/openhft/chronicle/bytes/Bytes.java:185]: why the { }

**Review by RobAustin — APPROVED** [review]

File: src/main/java/net/openhft/chronicle/bytes/Bytes.java (hunk 1 above)
Comment (minborg) [reply src/main/java/net/openhft/chronicle/bytes/Bytes.java:185]: Otherwise JavaDoc will get confused

File: pom.xml [hunk 2]
```diff
@@ -26,7 +26,7 @@
     </parent>
 
     <artifactId>chronicle-bytes</artifactId>
-    <version>2.22ea11-SNAPSHOT</version>
+    <version>2.22ea12-SNAPSHOT</version>
```
Comment (JerryShea) [inline pom.xml:29]: why is this changed?

File: src/main/java/net/openhft/chronicle/bytes/Bytes.java (hunk 1 above)
Comment (minborg) [reply src/main/java/net/openhft/chronicle/bytes/Bytes.java:185]: I think the `@FieldGroup`  interferes with the original `@code`

File: pom.xml (hunk 2 above)
Comment (minborg) [reply pom.xml:29]: `develop` moved on and I changed it when reproducing errors in Wire. I should have reverted it back before I created the PR.
